dist_run.py

```diff
@@ -231,6 +237,12 @@ def main():
     if pp_rank == pp_degree - 1 and tp_rank == 0:
         logger.info(f"Output: {output}")
 
+    # show peak memory stats for this stage
+    res_mem_gib, res_mem_pct = gpu_memory_monitor.get_peak_stats()
+    logger.info(
+        f"{color.blue} Memory used: {color.green}{res_mem_pct:.3f} %, {color.magenta}{res_mem_gib:.3f} GB{color.reset}"
+    )
+
```

Comment:
nit: maybe make the method "print_peak_stats()" if there is no other use of `res_mem_gib, res_mem_pct` than logging them?

Reply:
in this case, I wanted to return the data b/c if we want to setup tb logging or similar stress test monitoring, then we can pass this to summarywriter or other for dashboarding.